zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```
@@ -257,6 +271,9 @@ private[consumer] final class Runloop private (
                               for {
                                 ignoreRecordsForTps <- doSeekForNewPartitions(c, assignedTps)
 
+                                // invalidate current consumer group metadata
+                                _ <- consumerMetadataRef.set(None)
```

Can we immediately set the consumer group data here, instead of 'on-demand'?

---

🤔  We could, but there is no guarantee that the group metadata is needed (it is likely though). On the other hand it would remove a bit of complexity.

There is another aspect: method `getConsumerGroupMetadataIfAny` is written such that if the call fails, it returns `None`. That means that the call is retried every time. We lose the retries when the logic is moved to this place. I am still doubting between whether this is good or bad 🤷

---

I can do change, but I have limited understanding for the second aspect Erik shared. If we fail to set `getConsumerGroupMetadataIfAny`, would we retry the current poll and if so, would it be still marked as rebalance event?
If above happens as I said, it seems like a good change.

---

We will have to look in the git history to see why any error is being ignored from the call to `consumer.groupMetadata()`. I will have time for that on Wednesday but feel free to do this earlier. To be honest I don't expect to find an answer but we can try. Perhaps @iravid remembers?

On the other hand, as the PR is right now, it already solves issue #1431. So I am in favor of merging this now and create a new issue to explore the idea of moving the logic. @svroonland WDYT?